lib/workers/evm_server.rb

```diff
@@ -143,6 +144,20 @@ def monitor
     _log.info("Server Monitoring Complete - Timings: #{timings.inspect}") unless timings[:total_time] < ::Settings.server.server_log_timings_threshold.to_i_with_method
   end
 
+  def messaging_health_check
+    require 'manageiq-messaging'
+
+    _log.info("Initiating messaging health check")
+    broker = MiqQueue.messaging_client("health_check")
+    broker.publish_message(:service => "messaging-health-check", :message => "health check", :payload => {})
+    broker.subscribe_messages(:service => "messaging-health-check") { break }
+  rescue => err
+    _log.error("Messaging health check failed: #{err}")
+    kill_servers
```

Comment:
Wondering if it's better to `stop_servers` here, since killing the workers will only cause them to be restarted once again

Reply:
If we move this into `MiqServer` then we can use `shutdown_and_exit(1)` I think we want to take down the current evmserverd process so that the wait_for_messaging pre-script will kick in